src/simulation/Benchmark.cpp

```diff
@@ -0,0 +1,431 @@
+{
+    for (size_t accountsLeft = size, batchSize = size; accountsLeft > 0;
+         accountsLeft -= batchSize)
+    {
+        batchSize = std::min(accountsLeft, MAXIMAL_NUMBER_OF_TXS_PER_LEDGER);
+        auto newAccounts = sampler.createAccounts(batchSize);
+        createAccountsDirectly(app, newAccounts);
+    }
+}
+
+void
+Benchmark::BenchmarkBuilder::setMaxTxSize(LedgerManager& ledger,
+                                          uint32_t maxTxSetSize) const
+{
+    StellarValue sv(ledger.getLastClosedLedgerHeader().hash,
+                    ledger.getLedgerNum(), emptyUpgradeSteps, 0);
+    {
+        LedgerUpgrade up(LEDGER_UPGRADE_MAX_TX_SET_SIZE);
+        up.newMaxTxSetSize() = maxTxSetSize;
+        Value v(xdr::xdr_to_opaque(up));
+        sv.upgrades.emplace_back(v.begin(), v.end());
+    }
+    LedgerCloseData ledgerData = createData(ledger, sv);
+    ledger.closeLedger(ledgerData);
+}
+
+LedgerCloseData
+Benchmark::BenchmarkBuilder::createData(LedgerManager& ledger,
+                                        StellarValue& value) const
+{
+    auto ledgerNum = ledger.getLedgerNum();
+    TxSetFramePtr txSet =
+        std::make_shared<TxSetFrame>(ledger.getLastClosedLedgerHeader().hash);
+    value.txSetHash = txSet->getContentsHash();
+    return LedgerCloseData{ledgerNum, txSet, value};
+}
+
+void
+Benchmark::BenchmarkBuilder::createAccountsDirectly(
+    Application& app,
+    std::vector<LoadGenerator::AccountInfoPtr>& createdAccounts) const
+{
+    soci::transaction sqlTx(app.getDatabase().getSession());
+
+    auto ledger = app.getLedgerManager().getLedgerNum();
+
+    int64_t balanceDiff = 0;
+    std::vector<LedgerEntry> live;
+    std::transform(
+        createdAccounts.begin(), createdAccounts.end(),
+        std::back_inserter(live),
+        [&app, &balanceDiff](LoadGenerator::AccountInfoPtr const& account) {
+            AccountFrame aFrame = account->createDirectly(app);
+            balanceDiff += aFrame.getBalance();
+            return aFrame.mEntry;
+        });
+
+    SecretKey skey = SecretKey::fromSeed(app.getNetworkID());
+    AccountFrame::pointer masterAccount =
+        AccountFrame::loadAccount(skey.getPublicKey(), app.getDatabase());
+    LedgerDelta delta(app.getLedgerManager().getCurrentLedgerHeader(),
+                      app.getDatabase());
+    masterAccount->addBalance(-balanceDiff);
+    masterAccount->touch(ledger);
+    masterAccount->storeChange(delta, app.getDatabase());
+
+    sqlTx.commit();
+
+    auto liveEntries = delta.getLiveEntries();
+    live.insert(live.end(), liveEntries.begin(), liveEntries.end());
+    app.getBucketManager().addBatch(app, ledger, live, {});
+
+    StellarValue sv(app.getLedgerManager().getLastClosedLedgerHeader().hash,
+                    ledger, emptyUpgradeSteps, 0);
+    LedgerCloseData ledgerData = createData(app.getLedgerManager(), sv);
+    app.getLedgerManager().closeLedger(ledgerData);
```

Comment:
curious about this - does that actually work? Or you have to run it on a single node (and not on a node that has joined a network)?
I would imagine that this is competing with the node closing ledgers with the rest of the network (or even itself) and getting borked - so I imagine that this `createAccountsDirectly` can only be run when the app is in a very specific state (no force scp, node not in sync, no peers, etc).
 It seems that there is a very specific sequence of commands to run to make this work so we should probably create some documentation in `docs/benchmarking.md`